src/lib/version.cmake

```diff
@@ -0,0 +1,7 @@
+# create or update the version header file with the latest git describe
+execute_process(
+    COMMAND git describe --dirty
+    OUTPUT_VARIABLE GIT_DESCRIBE
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+configure_file(${SRC} ${DST} @ONLY)
```

Comment:
Hmm... it looks like maybe `CMAKE_SOURCE_DIR` and `CMAKE_BINARY_DIR` won't necessarily be passed thru.